### DISCUSSION TITLE
Your project's URL and Key are required to create a Supabase client! Vercel deployment preview

### DISCUSSION THREAD

**Discussion by Cosmonaut-itc** [supabase/supabase#22142] [Questions] [answered]
Using the new supabase integration I get this error when deploying previews to my vercel project:
`Error: either NEXT_PUBLIC_SUPABASE_URL and NEXT_PUBLIC_SUPABASE_ANON_KEY env variables or supabaseUrl and supabaseKey are required!`

I looked up here in the discussions and found a similar issue, here is the link: https://github.com/orgs/supabase/discussions/3218

The workaround stated did not work for me, as it kept happening only in my previews. When I deploy directly to production, it works as always, but as soon as I create a new branch and deploy a preview I cant get past this error. It only began happening in my previews after installing the new integration and removing the deprecated version.

Any Idea on how to fix it?

My next.js version is: `14.0.3`
My supabase version is this: `2.21.0`
I have not upgraded yet to the supabase ssr package and I am still using the `@supabase/auth-helpers-nextjs` in the "0.6.1" version, so my setup is using the previous recommended files and functions (supabase-listener,provider,browser,server). 
I have to point out I did not have any issue previously with the auth-helpers-nextjs package. 

I will greatly appreciate any suggestions you might have

UPDATE: I refactored my code to use supabase ssr and stopped using `@supabase/auth-helpers-nextjs`. I've fixed the issue where the preview was not deploying. They succesfully deploy now but now my middleware function is crashing with a similar error:

`Error: Your project's URL and Key are required to create a Supabase client!

Check your Supabase project's API settings to find these values

https://supabase.com/dashboard/project/_/settings/api
    at (node_modules/@supabase/ssr/dist/index.mjs:221:10)
    at (src/utils/supabase/middleware.js:14:21)
    at (src/middleware.js:6:17)
    at (node_modules/@sentry/nextjs/esm/common/utils/edgeWrapperUtils.js:54:42)
    at (node_modules/@sentry/nextjs/esm/common/wrapMiddlewareWithSentry.js:18:9)
    at (node_modules/next/dist/esm/server/web/adapter.js:152:22)
    at (node_modules/next/dist/esm/server/async-storage/request-async-storage-wrapper.js:72:23)
    at (node_modules/next/dist/esm/server/web/adapter.js:139:52)`

I have the same exact code that the next js docs have for the middleware and createClient. I will post it anyways in case someone spot something in them:

middleware.js:

`
import { updateSession } from "@/utils/supabase/middleware";

export async function middleware(request) {
    return await updateSession(request);
}

export const config = {
    matcher: [
        /*
         * Match all request paths except:
         * - _next/static (static files)
         * - _next/image (image optimization files)
         * - favicon.ico (favicon file)
         * - images - .svg, .png, .jpg, .jpeg, .gif, .webp
         * Feel free to modify this pattern to include more paths.
         */
        "/((?!_next/static|_next/image|favicon.ico|.*\\.(?:svg|png|jpg|jpeg|gif|webp)$).*)",
    ],
};

`

supabase/middleware.js:
`
import { createServerClient } from '@supabase/ssr'
import { NextResponse } from 'next/server'

export async function updateSession(request) {
    let response = NextResponse.next({
        request: {
            headers: request.headers,
        },
    })
    

    const supabase = createServerClient(
        process.env.NEXT_PUBLIC_SUPABASE_URL,
        process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY,
        {
            cookies: {
                get(name) {
                    return request.cookies.get(name)?.value
                },
                set(name, value, options) {
                    request.cookies.set({
                        name,
                        value,
                        ...options,
                    })
                    response = NextResponse.next({
                        request: {
                            headers: request.headers,
                        },
                    })
                    response.cookies.set({
                        name,
                        value,
                        ...options,
                    })
                },
                remove(name, options) {
                    request.cookies.set({
                        name,
                        value: '',
                        ...options,
                    })
                    response = NextResponse.next({
                        request: {
                            headers: request.headers,
                        },
                    })
                    response.cookies.set({
                        name,
                        value: '',
                        ...options,
                    })
                },
            },
        }
    )

    // refreshing the auth token
    await supabase.auth.getUser()

    return response
}

`

supabase/server.js

`
import { createServerClient } from '@supabase/ssr';
import { cookies } from 'next/headers';

export function createClient() {
    const cookieStore = cookies();

    return createServerClient(
        process.env.NEXT_PUBLIC_SUPABASE_URL,
        process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY,
        {
            cookies: {
                get(name) {
                    return cookieStore.get(name)?.value;
                },
                set(name, value, options) {
                    try {
                        cookieStore.set({ name, value, ...options });
                    } catch (error) {
                        // The `set` method was called from a Server Component.
                        // This can be ignored if you have middleware refreshing
                        // user sessions.
                    }
                },
                remove(name, options) {
                    try {
                        cookieStore.set({ name, value: '', ...options });
                    } catch (error) {
                        // The `delete` method was called from a Server Component.
                        // This can be ignored if you have middleware refreshing
                        // user sessions.
                    }
                },
            },
        }
    );
}

`

This issue only happens when deploying it on Vercel, in my dev environment it works perfectly and my `env` variables are working correctly

**Comment by Nunyaklah** [comment, marked as answer]
Is it possible that your `env` variables are unavailable on the new branches you create for previews? Because you mentioned it does not happen in production

**Reply by Cosmonaut-itc** [reply]
I am not sure, I did a test and if I deploy the exact same page I have in production into a preview branch I still get the error, before installing the new integrations I never had an issue with previews so I don't know if I need extra configuration or if the vercel previews get the `env` variables from different declarations other than `NEXT_PUBLIC`

**Reply by Cosmonaut-itc** [reply]
You were right @Nunyaklah the integration makes the `env` variables only available to production on a fresh install. Thanks for the help :D

**Reply by Nunyaklah** [reply]
That's great news, you welcome :)

**Reply by guipace** [reply]
I'm having this exact issue after adding the supabase integration. Did you guys solve it by removing the integration and manually adding the env vars or is there a way to make it work maintaining the integration?

**Reply by Cosmonaut-itc** [reply]
> I'm having this exact issue after adding the supabase integration. Did you guys solve it by removing the integration and manually adding the env vars or is there a way to make it work maintaining the integration?

Hey @guipace you don't need to remove the integration. It's annoying but I discovered that the integration only makes available the `env` variables in production. You need to go to your Vercel Project settings -> Environmental Variables and click on the options of the supabase anon Key, supabase URL and manually turn them on for previews. The deprecated integration did that for you so it can be easily overlooked

**Reply by guipace** [reply]
Ah, thanks for pointing out the obvious! I hadn't seen those details and the ability to edit the env var to add more environments. New to both Vercel and Supabase. Appreciate it!
